artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerSource.java

```diff
@@ -213,15 +215,31 @@ public void deleteQueue(SimpleString address, SimpleString queue) throws Excepti
 
       if (deleteQueues) {
          Message message = createMessage(address, queue, DELETE_QUEUE, null, queue.toString());
-         route(server, message);
+         routeMirrorCommand(server, message);
       }
    }
 
+   private boolean invalidTarget(MirrorController controller, Message message) {
+      if (controller == null) {
+         return false;
+      }
+      String remoteID = getRemoteMirrorId();
+      if (remoteID == null) {
+         // This is to avoid a reflection from a small period of time one node reconnects but not the opposite direction
+         remoteID = String.valueOf(message.getAnnotation(BROKER_ID_SIMPLE_STRING));
+         logger.debug("Remote link is not initialized yet, setting remoteID from message as {}", remoteID);
```

Comment:
In the short period span the remoteID could be null, and if the message has no annotation, it would receive "null", and the check if the same ID would fail as expected.

I could make it more intentional by checking though.